docs/api-doc.md

```diff
@@ -0,0 +1,160 @@
+
+## Sass Context Internals
+
+Everything is stored in structs:
+
+```C
+struct Sass_Options;
+struct Sass_Context : Sass_Options;
+struct Sass_File_context : Sass_Context;
+struct Sass_Data_context : Sass_Context;
+```
+
+This mirrors very well how `libsass` uses these structures.
+
+- `Sass_Options` holds everything you feed in before the compilation. It also hosts `input_path` and `output_path` options, because they are used to generate/calculate relative links in source-maps. The `input_path` is shared with `Sass_File_Context`.
+- `Sass_Context` holds all the data returned by the compilation step. 
+- `Sass_File_Context` is a specific implementation that requires no additional fields
+- `Sass_Data_Context` is a specific implementation that adds the `input_source` field
+
+Structs can be down-casted to access `context` or `options`!
+
+## Common Pitfalls
+
+**input_path**
+
+The `input_path` is part of `Sass_Options`, but it also is the main option for `Sass_File_Context`. It is also used to generate relative file links in source-maps. Therefore it is pretty usefull to pass this information if you have a `Sass_Data_Context` and know the original path.
+
+**output_path**
+
+Be aware that `libsass` does not write the output file itself. This option merely exists to give `libsass` the proper information to generate links in source-maps. The file has to be written to the disk by the binding/implementation. If the `output_path` is omitted, `libsass` tries to extrapolate one from the `input_path` by replacing (or adding) the file ending with `.css`.
+
+## Error Codes
+
+The `error_code` is integer value which indicates the type of error that occurred inside the libsass process. Following is the list of error codes along with the short description:
+
+* 1: normal errors like parsing or `eval` errors
+* 2: bad allocation error (memory error)
+* 3: "untranslated" C++ exception (`throw std::exception`)
+* 4: legacy string exceptions ( `throw const char*` or `std::string` )
+* 5: Some other unknown exception
+
+Although for the API consumer, error codes do not offer much value except indicating whether *any* error occurred during the compilation, it helps debugging the libsass internal code paths.
+
+## Real-World Implementations
+
+The proof is in the pudding, so we have highlighted a few implementations that should be on par with the latest libsass interface version. Some of them may not have all features implemented!
+
+1. [Perl Example](https://github.com/sass/perl-libsass/blob/master/lib/CSS/Sass.xs)
+2. [Go Example](http://godoc.org/github.com/wellington/go-libsass#example-Context-Compile)
+3. [Node Example](https://github.com/sass/node-sass/blob/master/src/binding.cpp)
+
+## ABI forward compatibility
+
+We use a functional API to make dynamic linking more robust and future compatible. The API is not yet 100% stable, so we do not yet guarantee ABI forward compatibility. We will do so, once we increase the shared library version above 1.0.
```

Comment:
Can you link to the API here so people can read up on it for further details if they find a problem?